src/Tests/VirtualMachine.fs

```diff
@@ -136,4 +137,17 @@ let tests = testList "Virtual Machine" [
         Expect.equal keyData "fooKey1" "public keys were not correctly added"
         Expect.equal path "fooPath1" "path was not correctly added"
     }
+
+    test "Handles identity correctly" {
+        let machine = arm { add_resource (vm { name ""; username "isaac" }) } |> findAzureResources<VirtualMachine> client.SerializationSettings |> Seq.head
+        Expect.isNull machine.Identity "Default managed identity should be null"
+
+        let machine = arm { add_resource (vm { system_identity; username "isaac" }) } |> findAzureResources<VirtualMachine> client.SerializationSettings |> Seq.head
+        Expect.equal machine.Identity.Type (Nullable ResourceIdentityType.SystemAssigned) "Should have system identity"
+        Expect.isNull machine.Identity.UserAssignedIdentities "Should have no user assigned identities"
+
+        let machine = arm { add_resource (vm { system_identity; add_identity (createUserAssignedIdentity "test"); add_identity (createUserAssignedIdentity "test2"); username "isaac" }) } |> findAzureResources<VirtualMachine> client.SerializationSettings |> Seq.head
```

Comment:
Linking to an existing UAI - good point, I don't think we have support for that. Can add that as a separate thing though?

> I'm not sure this usage will really work in practice because the user assigned identity must also be added to the template. It works for a test, but I usually need to create them separately so I can add them to both the template and the resource.

What line are you referring to here - it's not clear from the code sample?

Reply:
ah. this is just for the purposes of the unit test. happy to move it out separately if that improves readability.

Reply:
Linking to #672

Reply:
Referring to your earlier comment - I actually think that if you simply create a managed user identity and *don't* add it to Farmer, it effectively acts as exactly what you're referring to (I think).